infrastructure/code/code.go

```diff
@@ -116,10 +116,26 @@ func (sc *Scanner) SupportedCommands() []snyk.CommandName {
 }
 
 func (sc *Scanner) Scan(ctx context.Context, path string, folderPath string) (issues []snyk.Issue, err error) {
-	if !sc.isSastEnabled() {
+	sastResponse, err := sc.SnykApiClient.SastSettings()
+	method := "Scan"
+
+	if err != nil {
+		log.Error().Err(err).Str("method", method).Msg("couldn't get sast enablement")
+		sc.errorReporter.CaptureError(err)
+		return issues, errors.New("couldn't get sast enablement")
+	}
+
+	if !sc.isSastEnabled(sastResponse) {
 		return issues, errors.New("SAST is not enabled")
 	}
 
+	if sc.isLocalEngineEnabled(sastResponse) {
+		isCodeApiUpdated := sc.updateCodeApiLocalEngine(sastResponse)
+		if !isCodeApiUpdated {
+			return issues, errors.New("Local engine not configured correctly.")
```

Comment:
This should be logged with `sc.errorReporter.CaptureError(err)`

Reply:
Should `SAST is not enabled` be logged in the same way? 

I don't think so, because having SAST disabled is not an actual error.